File: 10-6.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace std;
using namespace cv;

int main(){

    Mat img_color = imread("house.png");

    Mat img_gray;
    cvtColor(img_color, img_gray, COLOR_BGR2GRAY);
    imshow("Original", img_color);

    Mat img_edge;
    blur(img_gray, img_edge, Size(3,3));

    Canny(img_edge, img_edge, 50,150,3);

    imshow("Canny Edge", img_edge);
    waitKey(0);

    destroyAllWindows();
    return 0;
}

File: 10-6.py
import cv2 as cv

img_gray = cv.imread('house.png', cv.IMREAD_GRAYSCALE);
cv.imshow('Original', img_gray);

img_gray = cv.blur(img_gray, (3,3))
img_canny = cv.Canny(img_gray, 50,150)  
cv.imshow('Canny Edge', img_canny)

cv.waitKey(0)
cv.destroyAllWindows()